feat(curriculum): add lunch picker lab

zairahira opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

I have modified the original prototype a bit as it was challenging to test constantly updating arrays.

<!-- Feel free to add any additional description of changes below this line -->

zairahira committed:
chore: add project structure

zairahira committed:
feat: add stories and tests

zairahira committed:
Merge remote-tracking branch 'upstream/main' into feat/lunch-picker-lab

zairahira committed:
Revert extra file to its original state

jdwilkin4 commented on client/i18n/locales/english/intro.json:
```diff
@@ -1967,6 +1967,12 @@
+      "lab-lunch-picker-program": {
+        "title": "Build a Lunch Picker Program",
+        "intro": [
+          "In this lab, you create a lunch picker user arrays and random numbers"
```
```suggestion
          "In this lab, you will review working with arrays and random numbers by building a lunch picker program."
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+
+# --description--
+
+In this lab, you will create a program that will help you decide what to eat for lunch. You will have two arrays with different food items. You will use the `push`, `pop` and `shift` methods to modify the arrays. You will also use the `Math.random()` method to generate a random number that will be used to select a random meal from the arrays.
```
I think this description gives away to much. We should have campers think a little about what methods they will need to use.

```suggestion
In this lab, you'll build a program to assist in choosing lunch options. You'll work with two arrays: one for children's food choices and one for adults. You'll add and remove items from these arrays and then randomly select a meal from each. Finally, you'll log the results to the console.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+
+**User Stories:**
+
+1. You should create a `kidsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
```
we shouldn't mention how they should go about it. it is best to have campers think a little about what method or approach should work here.

```suggestion
1. You should create a `kidsMeal` variable and assign it an array with at least three strings in it.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+**User Stories:**
+
+1. You should create a `kidsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+2. You should create an `adultsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
```
```suggestion
2. You should create an `adultsMeal` variable and assign it an array with at least three strings in it.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+
+1. You should create a `kidsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+2. You should create an `adultsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+3. Once items have been added to the `kidsMeal`, you should remove the last item from it using `pop()`.
```
```suggestion
3. Once items have been added to the `kidsMeal`, you should use the appropriate array method to remove the last item.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+1. You should create a `kidsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+2. You should create an `adultsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+3. Once items have been added to the `kidsMeal`, you should remove the last item from it using `pop()`.
+4. Once items have been added to the `adultsMeal`, you should remove the first item from it using `shift()`.
```
```suggestion
4. Once items have been added to the `adultsMeal`, you should use the appropriate array method to remove the first item. 
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+2. You should create an `adultsMeal` array and assign more than two string values to it. You can either use `push()` or assignment to add the items to the array.
+3. Once items have been added to the `kidsMeal`, you should remove the last item from it using `pop()`.
+4. Once items have been added to the `adultsMeal`, you should remove the first item from it using `shift()`.
+5.  Generate a random number between `0` to the length of the `kidsMeal` array using `Math.random()` and store it in a variable called `randomIndexKids`.
```
```suggestion
5.  You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexKids`.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,149 @@
+3. Once items have been added to the `kidsMeal`, you should remove the last item from it using `pop()`.
+4. Once items have been added to the `adultsMeal`, you should remove the first item from it using `shift()`.
+5.  Generate a random number between `0` to the length of the `kidsMeal` array using `Math.random()` and store it in a variable called `randomIndexKids`.
+6. Generate a random number between `0` to the length of the `adultsMeal` array using `Math.random()` and store it in a variable called `randomIndexAdults`.
```
```suggestion
6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexAdults`.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,149 @@
+Your `kidsMeal` array should contain atleast three items.
+
+```js
+assert.isAtLeast(kidsMeal.length, 2);
````
```suggestion
assert.isAtLeast(kidsMeal.length, 3);
```

zairahira replied:
I have kept this intentionally as this would be the length after using `pop()` and `shift()` on the respective arrays with length `3`.

Dario-DC replied:
This should check that the array contains at least 3 elements at the moment of the assignment. Since it is modified, I'm afraid we should use regex for this test. Something similar to the following?

```suggestion
const matched = code.match(/(const|let)\s+kidsMeal\s*=\s*\[(?<items>.+?)\]/)
const kidsMealArray = matched.groups.items.split(",");
assert.isAtLeast(kidsMealArray.length, 3);
```

ilenia-magoni replied:
to avoid any issues with comments I think they should be removed any time a regex test is used

Dario-DC replied:
Suggestion updated following your comment @ilenia-magoni

zairahira replied:
Thank you @Dario-DC for the suggestion, this works perfectly.
I have updated other tests too.

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,149 @@
+Your `adultsMeal` array should contain atleast three items.
+
+```js
+assert.isAtLeast(adultsMeal.length, 2);
````
```suggestion
assert.isAtLeast(adultsMeal.length, 3);
```

zairahira replied:
I have kept this intentionally as this would be the length after using `pop()` and `shift()` on the respective arrays with length `3`.

zairahira replied:
Now, this test has nee updated.

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,149 @@
+assert.isAtLeast(adultsMeal.length, 2);
+```
+
+Your adultsMeal array should contain only string values.
````
```suggestion
Your `adultsMeal` array should contain only string values.
```

jdwilkin4 reviewed (commented):
Left some comments so the campers have to think more about the appropriate methods to use instead of telling them right off 👍

zairahira committed:
Apply suggestions from code review

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

zairahira committed:
cleanup

zairahira committed:
Merge remote-tracking branch 'upstream/main' into feat/lunch-picker-lab

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -41,7 +41,7 @@ assert.isArray(adultsMeal);
 Your `kidsMeal` array should contain atleast three items.
```
```suggestion
Your `kidsMeal` array should contain at least three items.
```

jdwilkin4 replied:
@zairahira it looks like we still have this last suggested change

zairahira committed:
fix: update tests

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,157 @@
+Your `kidsMeal` array should contain only non-empty string values.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+adultsMeal\s*=\s*\[(?<items>.+?)\]/)
+const adultsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(adultsMealArray.length, 1); 
+
+for (let i = 0; i < kidsMeal.length; i++) {
+  assert.isString(kidsMeal[i]); // Check that every item is a string
+  assert.isAbove(kidsMeal[i].length, 0, 'String should not be empty'); // Check that string is not empty
+}
````
The test is broken (referencing `adultsMeal`) but it doesn't work because the match is always a string. The following should work well.
The problem is if the items are variables. For that we need a different solution.
```suggestion
const matched = __helpers.removeJSComments(code).match(/(const|let)\s+kidsMeal\s*=\s*\[(?<items>.+?)\]/)
const kidsMealArray = matched.groups.items.split(",");
assert.isAtLeast(kidsMealArray.length, 1);

for (let i = 0; i < kidsMealArray.length; i++) {
  const itemMatch = kidsMealArray[i].trim().match(/^(?<quote>'|"|`)(?<meal>.+)\1$/);
  assert.isNotNull(itemMatch);
  assert.notInclude(itemMatch.groups.meal, itemMatch.groups.quote)
}
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,157 @@
+```
+
+You should use `Math.random()` to generate a random number from `0` upto the length of the `kidsMeal` array that will be used to select a random meal from the `kidsMeal` array and store it in a variable called `randomIndexKids`.
+
````
This test doesn't work properly. If I have `const randomIndexKids = Math.floor(Math.random() * 10)`, and when the test runs `randomIndexKids` is `1`, the  test does not fail. If you run it again it could be `9` and fail.

Also, it matches `Math.random()` without any relation with the assignment to `randomIndexKids`.

The same is true for the test after this one.

zairahira replied:
I have updated both, let me know what you think @Dario-DC.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,157 @@
+const adultsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(adultsMealArray.length, 1);
+
+for (let i = 0; i < adultsMeal.length; i++) {
+  assert.isString(adultsMeal[i]); 
+  assert.isAbove(adultsMeal[i].length, 0, 'String should not be empty'); 
+}
```
```suggestion

for (let i = 0; i < adultsMealArray.length; i++) {
  const itemMatch = adultsMealArray[i].trim().match(/^(?<quote>'|"|`)(?<meal>.+)\1$/);
  assert.isNotNull(itemMatch);
  assert.notInclude(itemMatch.groups.meal, itemMatch.groups.quote)
}
```

zairahira committed:
fix spelling

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

zairahira committed:
Apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira committed:
modify tests for random numbers

Dario-DC reviewed (commented):
I don't like the idea of using regex too much, but in this case I think we have to. I left a few comments to take spacing into account.

zairahira committed:
Apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira commented:
Thank you for refining the regex for these tests @Dario-DC 👍️

jdwilkin4 reviewed (commented):
I feel like we are spending a lot of time trying to get the regex tests to work.

But have you considered rewriting this to use functions instead? 
that way, you can test the final outcomes and test the functions again many arrays instead of just one

I just feel like we should look at other options that will make testing this easier instead of just sticking to this one version of the prototype. 

Also, I don't know if we need a user story to specifically test for the console statements. maybe just having them focus on the core functionality is enough.

Dario-DC commented:
I agree with Jessica. I didn't realize that this lab was after the function block. But since functions are already taught it would be better to refactor the code to use a function.

zairahira commented:
I have revamped the project to now use functions.
@Dario-DC and @jdwilkin4, let me know what you think.

If this seems fine, I'll start writing the tests.

Updated code:

```js
const lunches = [];

// Add a new lunch to the list (at the end)
function addLunchToEnd(newLunch) {
  lunches.push(newLunch);
  console.log(`${newLunch} added to the end of the lunch menu.`);
}

// Add a new lunch to the start of the list
function addLunchToStart(newLunch) {
  lunches.unshift(newLunch);
  console.log(`${newLunch} added to the start of the lunch menu.`);
}

// Remove the last lunch from the list
function removeLastLunch() {
  const removedLunch = lunches.pop();
  if (removedLunch) {
    console.log(`${removedLunch} removed from the lunch menu (last).`);
  } else {
    console.log("No lunches to remove.");
  }
}

// Remove the first lunch from the list
function removeFirstLunch() {
  const removedLunch = lunches.shift();
  if (removedLunch) {
    console.log(`${removedLunch} removed from the lunch menu (first).`);
  } else {
    console.log("No lunches to remove.");
  }
}

// Function to get a random lunch from the list
function getRandomLunch() {
  if (lunches.length === 0) {
    console.log("No lunches available.");
    return;
  }
  const randomIndex = Math.floor(Math.random() * lunches.length);
  const randomLunch = lunches[randomIndex];
  console.log(`Randomly selected lunch: ${randomLunch}`);
}

// New function to display the lunches as a numbered menu using a for loop
function showLunchMenu() {
  if (lunches.length === 0) {
    console.log("The menu is empty.");
    return;
  }
  console.log("Lunch Menu:");
  for (let i = 0; i < lunches.length; i++) {
    console.log(`${i + 1}. ${lunches[i]}`);
  }
}

// Example usage:
addLunchToEnd("Tacos");
addLunchToStart("Sushi");
removeLastLunch();
removeFirstLunch();
getRandomLunch();
showLunchMenu();

```

Updated user stories:

```md
# --description--

In this lab, you'll build a program to assist in managing lunch options. You'll work with an array of lunches, add and remove items from the array, and randomly select a lunch option. 

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should create a `lunches` variable and assign it some string values of different lunch items.
2. You should create a function `addLunchToEnd` that adds a new lunch item to the end of the `lunches` array.
3. You should create a function `addLunchToStart` that adds a new lunch item to the start of the `lunches` array.
4. You should create a function `removeLastLunch` that removes the last lunch item from the `lunches` array using the appropriate array method.
5. You should create a function `removeFirstLunch` that removes the first lunch item from the `lunches` array using the appropriate array method.
6. You should create a function `getRandomLunch` that generates a random number between `0` and the length of the `lunches` array. Use that random number to select a lunch and print it to the console.
7. You should create a function `showLunchMenu` that displays the current lunches as a numbered list in the format:
    - `"1. <Lunch Item>"`
    - `"2. <Lunch Item>"` (and so on, for each lunch in the array).


```

ilenia-magoni commented:
with the new version using `Math.random` you can test with the new helper that makes `Math.random` gives a series of numbers (removes the randomness) https://github.com/freeCodeCamp/curriculum-helpers/pull/233.

If the return statement has already been introduced at this point I would suggest having functions that return things instead of having them only print to the console

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,153 @@
+3. Once items have been added to the `kidsMeal`, you should use the appropriate array method to remove the last item.
+4. Once items have been added to the `adultsMeal`, you should use the appropriate array method to remove the first item. 
+5.  You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexKids`.
+6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexAdults`.
```
```suggestion
6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `adultsMeal` array. Assign that result to a variable called `randomIndexAdults`.
```

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,153 @@
+4. Once items have been added to the `adultsMeal`, you should use the appropriate array method to remove the first item. 
+5.  You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexKids`.
+6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexAdults`.
+7.  Log the following messages to the console:
```
```suggestion
7.  You should log the following messages to the console:
```

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,153 @@
+6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexAdults`.
+7.  Log the following messages to the console:
+
+- `"Today kids should eat:"` followed by the selected meal to the console.
+- `"Today adults should eat:"` followed by the selected meal to the console.
```
I worry a little that people will think they need to log the quotes to the console - but I'm fine with it like this for now. If people are getting confused, maybe we can change it later.

moT01 reviewed (changes requested):
This looks good @zairahira 👍 

Found one typo in the user stories, and left one other grammar suggestion.

jdwilkin4 commented:
@zairahira 

The updated project looks good.
I do agree with ilenia though, that we should return values instead of logging them to the console

zairahira committed:
Merge remote-tracking branch 'upstream/main' into feat/lunch-picker-lab

zairahira committed:
chore: add updated user stories and solution

zairahira committed:
chore: add updated hints

zairahira committed:
add new tests

zairahira committed:
cleanup

jdwilkin4 commented:
> There is a test to check the results of getRandomLunch, would another test be needed?

I personally vote we can leave it as is for now.
We will need to do more dedicated testing for all labs and workshops in the beta phase anyway and we will be able to see if more tests are needed anyway.

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,269 @@
+4. You should create a function `removeLastLunch` that removes the last lunch item from the `lunches` array and:
+   
+   - If successful, returns a string in the format:  
+     - `"[Lunch Item] removed from the lunch menu (last)."`
```
```suggestion
     - `"[Lunch Item] removed from the end of the lunch menu."`
```

This sounded a little off to me - but not really an issue to leave it, so no need to make the change if you don't want. The test would have to change as well.

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
```diff
@@ -0,0 +1,269 @@
+5. You should create a function `removeFirstLunch` that removes the first lunch item from the `lunches` array and:
+   
+   - If successful, returns a string in the format:  
+     - `"[Lunch Item] removed from the lunch menu (first)."`
```
```suggestion
     - `"[Lunch Item] removed from the start of the lunch menu."`
```

Same - This sounded a little off to me - but not really an issue to leave it, so no need to make the change if you don't want. The test would have to change as well.

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,269 @@
+assert.match(addLunchToEnd.toString(), /\s*addLunchToEnd\(\s*\w+\s*\)/);
+```
+
+`addLunchToEnd("Tacos)` should return the string `"Tacos added to the end of the lunch menu."`.
````
```suggestion
`addLunchToEnd("Tacos")` should return the string `"Tacos added to the end of the lunch menu."`.
```

moT01 reviewed (commented):
This was fun @zairahira 🎉 

I left a few minor suggestions - let me know if you just want to leave those as-is. 

<details><summary>One other thing that bothered me a little was all the nesting of lists in the user stories:</summary>

<img width="846" alt="Screenshot 2024-09-25 at 11 36 56 AM" src="https://github.com/user-attachments/assets/f987561d-2150-4ece-8b81-df98d00f5275">

</details>

 <details><summary> I played with it a little and came up with this:</summary>

<img width="894" alt="Screenshot 2024-09-25 at 5 53 31 PM" src="https://github.com/user-attachments/assets/6a2ecfb3-ac50-4df6-b8e4-f558730b6e88">

</details>

Not really sure if that looks any better TBH, so I would be fine to leave it.

zairahira committed:
apply suggestion from code review

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

zairahira commented:
@moT01, the nested lists look better readability wise. If you have the markdown, can you share it with me? You can also commit directly if you like.

moT01 commented:
Here's the markdown that produces the image @zairahira:

<details><summary>code</summary>

```
1. You should create a `lunches` variable and assign it an empty array that will be used to store lunch items.

2. You should create a function `addLunchToEnd` that takes a string parameter and adds it to the end of the `lunches` array and returns a string in the format: `"[Lunch Item] added to the end of the lunch menu."`

3. You should create a function `addLunchToStart` that takes a string parameter and adds it to the start of the `lunches` array and returns a string in the format: `"[Lunch Item] added to the start of the lunch menu."`

4. You should create a function `removeLastLunch` that removes the last lunch item from the `lunches` array and:
   - If successful, returns a string in the format: `"[Lunch Item] removed from the lunch menu (last)."`
   - If the `lunches` array is empty, returns a string: `"No lunches to remove."`

5. You should create a function `removeFirstLunch` that removes the first lunch item from the `lunches` array and:
   - If successful, returns a string in the format: `"[Lunch Item] removed from the lunch menu (first)."`
   - If the `lunches` array is empty, returns a string: `"No lunches to remove."`

6. You should create a function `getRandomLunch` that selects a random lunch item from the `lunches` array and:
   - If successful, returns a string in the format: `"Randomly selected lunch: [Lunch Item]"`
   - If the `lunches` array is empty, returns a string: `"No lunches available."`

7. You should create a function `showLunchMenu` that:
   - If there are items in the `lunches` array, returns a string in the format: `"Menu items: [Lunch Item], [Lunch Item]...`
   - If the `lunches` array is empty, returns a string: `"The menu is empty."`
```

</details>

zairahira committed:
chore: refactor use stories

zairahira commented:
The user stories have been updated and the changes have been reflected in hints, tests and solution.

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,253 @@
+
+When `lunches = ["Fish", "Fries", "Roast"]`, the function `getRandomLunch` should return a string in the format `"Randomly selected lunch: [Lunch Item]"`.
+
+```js
+let initialCondition = "false";
+lunches = ["Fish", "Fries", "Roast"]
+
+lunches.forEach((lunch) => {
+  const result = getRandomLunch();
+  if(result === `Randomly selected lunch: ${lunch}`){
+    initialCondition = "true";
+  }
+});
+assert(initialCondition);
+```
````
this test always passes, you have `assert("any string")` which always passes.
You need to convert `initialCondition` to a boolean, but then it can also randomly not pass.

I suggest using the new mockRandom thing so you know which item should come out, and you know which item should come out of `getRandomLunch`. Or you can even force `math.random` to come out with a specific value adding before the call to `getRandomLunch` one of these:
```js
math.random = () => 0; //to get the first item in the array
math.random = () => 0.4; // to get the second item in the array
math.random = () => 0.8; // to get the third item in the array
```

ilenia-magoni replied:
```suggestion
const temp = Math.random;

lunches = ["Fish", "Fries", "Roast"];

// check that it correctly outputs the first item
Math.random = () => 0;
assert.equal(getRandomLunch(), `Randomly selected lunch: ${lunches[0]}`);

// second item
Math.random = () => 0.4;
assert.equal(getRandomLunch(), `Randomly selected lunch: ${lunches[1]}`);

// third item
Math.random = () => 0.8;
assert.equal(getRandomLunch(), `Randomly selected lunch: ${lunches[2]}`);

// restore Math.random
Math.random = temp;
```

zairahira committed:
update test

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,261 @@
+assert.isFunction(addLunchToStart);
+```
+
+The function `addLunchToStart` should take a string as an argument.
````
```suggestion
The function `addLunchToStart` should take a single argument.
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md:
````diff
@@ -0,0 +1,261 @@
+The function `addLunchToStart` should take a string as an argument.
+
+```js
+assert.match(addLunchToStart.toString(), /\s*addLunchToStart\(\s*\w+\s*\)/);
````
```suggestion
assert.lengthOf(addLunchToStart, 1);
```

Dario-DC reviewed (commented):
The lab is looking good :+1: I left a few comments.

zairahira committed:
Apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

moT01 reviewed (approved):
LGTM 🎉
